pkg/expression/constant_propagation.go

```diff
@@ -242,11 +247,8 @@ var propConstSolverPool = sync.Pool{
 
 type propConstSolver struct {
 	basePropConstSolver
-	conditions []Expression
-	// When performing constant propagation, if the newly created expression cannot be pushed down,
-	// we might consider this expression to be invalid. We use the pushDownFilter here to determine
-	// whether it can be pushed down.
-	pushDownfilter func(expr Expression) bool
+	conditions    []Expression
+	vaildExprFunc VaildConstantPropagationExpressionFuncType
```

Comment:
better add a todo here, to eliminate func pointer

Reply:
I have updated.